src/zarr/codecs/pipeline.py

```diff
@@ -371,12 +377,16 @@ async def _read_key(byte_setter: ByteSetter | None) -> Buffer | None:
 
             def _merge_chunk_array(
```

Comment:
Does this function need to be defined inside the `write_batch` function? I noticed that `_merge_chunk_array` has access to `drop_axis`, even though `drop_axis` is not passed as an argument to `_merge_chunk_array`, and it took me a bit to notice that this is because `_merge_chunk_array` is defined in the body of another function where `drop_axis` is in scope. To keep things less surprising, maybe `_merge_chunk_array` should be defined at the same scope as all the other functions?